pingora-core/src/services/listening.rs

```diff
@@ -163,14 +163,19 @@ impl<A: ServerApp + Send + Sync + 'static> Service<A> {
             };
             match new_io {
                 Ok(io) => {
+                    let peer_addr = io.peer_addr();
                     let app = app_logic.clone();
                     let shutdown = shutdown.clone();
                     current_handle().spawn(async move {
                         match io.handshake().await {
                             Ok(io) => Self::handle_event(io, app, shutdown).await,
                             Err(e) => {
-                                // TODO: Maybe IOApp trait needs a fn to handle/filter our this error
-                                error!("Downstream handshake error {e}");
+                                // TODO: Maybe IOApp trait needs a fn to handle/filter out this error
+                                if let Some(addr) = peer_addr {
+                                    error!("Downstream handshake error from {}: {e}", addr);
```

Comment:
Getting the `peer_addr` incurs a `getpeername` syscall. Preferably you would do something like
```
error!("Downstream handshake error{}: {e}", io.peer_addr().map(|addr| format!(" from {}", addr.to_string())));
```
So that the syscall is only invoked when the error message is logged.